shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/surveyselection/SurveySelectionViewModel.kt

```diff
@@ -42,7 +52,41 @@ class SurveySelectionViewModel(
                 .combine(getAppVersion()) { userDate, version ->
                     Triple(userDate.first, userDate.second, version)
                 }
-                .collect { updateHeaderState(it.first, it.second, it.third) }
+                .combine(fetchSurvey(currentPage)) { userData, survey ->
+                    Pair(userData, survey)
+                }
+                .collect {
+                    updateHeaderState(it.first.first, it.first.second, it.first.third)
+                    updateSurveyState(it.second)
+                }
+        }
+    }
+
+    fun checkFetchMore(itemIndex: Int) {
+        if(itemIndex >= viewState.value.surveys.size - FETCH_MORE_TRIGGER) {
```

Comment:
Just curious why `FETCH_MORE_TRIGGER` is `2` instead of `1`? Don't we want to trigger to load more when we reach the last page so that the content is loaded in the background a bit faster, creating a more seamless experience, no? 🤔

Reply:
> content is loaded in the background a bit faster

Using 2 here would make the load faster isn't it? I use 2 here for the UX like you mention 😄 (2nd last vs last).

Reply:
Optional: we could also mention this optimization as a comment, as it is not very clear from the beginning why we go with the value `2` 😁